src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -223,5 +223,42 @@ public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
 
         /// <summary>Gets whether the system is case-sensitive.</summary>
         internal static bool IsCaseSensitive { get { return false; } }
+
+        internal static ReadOnlySpan<char> GetVolumeName(ReadOnlySpan<char> path)
+        {
+            if (!IsPathFullyQualified(path))
+                return default;
+
+            // 3 cases: UNC ("\\server\share"), Device ("\\?\C:\"), or Dos ("C:\")
+            ReadOnlySpan<char> root = GetPathRoot(path);
+            if (!PathInternal.IsDevice(path))
+                return TrimEndingDirectorySeparator(root); // e.g. "C:"
+
+            if (IsUnc(path))
```

Comment:
Need `if (IsUnc(path))` then a check for `IsDevicePath()` to handle both UNC cases.